cardano-db-sync/src/Cardano/DbSync/Sync.hs

```diff
@@ -137,43 +138,54 @@ runSyncNode ::
-  logInfo trce $ "Using alonzo genesis file from: " <> (show . unGenesisFile $ dncAlonzoGenesisFile enc)
+runSyncNode metricsSetters trce iomgr aop snEveryFollowing snEveryLagging dbConnString ranAll runMigration syncNodeParams = do
+  let configFile = enpConfigFile syncNodeParams
+      maybeLedgerDir = enpMaybeLedgerStateDir syncNodeParams
+  syncNodeConfig <- readSyncNodeConfig configFile
+  whenJust maybeLedgerDir $
+    \enpLedgerStateDir -> do
+      createDirectoryIfMissing True (unLedgerStateDir enpLedgerStateDir)
+
+  logInfo trce $ "Using byron genesis file from: " <> (show . unGenesisFile $ dncByronGenesisFile syncNodeConfig)
+  logInfo trce $ "Using shelley genesis file from: " <> (show . unGenesisFile $ dncShelleyGenesisFile syncNodeConfig)
+  logInfo trce $ "Using alonzo genesis file from: " <> (show . unGenesisFile $ dncAlonzoGenesisFile syncNodeConfig)
 
   orDie renderSyncNodeError $ do
-    genCfg <- readCardanoGenesisConfig enc
+    genCfg <- readCardanoGenesisConfig syncNodeConfig
     logProtocolMagicId trce $ genesisProtocolMagicId genCfg
     syncEnv <-
       ExceptT $
         mkSyncEnvFromConfig
           trce
           dbConnString
-          (SyncOptions (enpExtended enp) aop (enpHasCache enp) (enpHasLedger enp) (enpSkipFix enp) (enpOnlyFix enp) snEveryFollowing snEveryLagging)
-          (enpLedgerStateDir enp)
+          (SyncOptions
+            (enpExtended syncNodeParams)
+            aop
+            (enpHasCache syncNodeParams)
+            (enpHasLedger syncNodeParams)
+            (enpSkipFix syncNodeParams)
+            (enpOnlyFix syncNodeParams)
+            snEveryFollowing
+            snEveryLagging)
+          (enpMaybeLedgerStateDir syncNodeParams)
           genCfg
           ranAll
-          (enpForceIndexes enp)
+          (enpForceIndexes syncNodeParams)
           runMigration
 
     -- If the DB is empty it will be inserted, otherwise it will be validated (to make
     -- sure we are on the right chain).
     lift $ Db.runIohkLogging trce $ withPostgresqlConn dbConnString $ \backend -> do
-      liftIO $ unless (enpHasLedger enp) $ do
+      liftIO $ unless (enpHasLedger syncNodeParams) $ do
         logInfo trce "Migrating to a no ledger schema"
         Db.noLedgerMigrations backend trce
-      lift $ orDie renderSyncNodeError $ insertValidateGenesisDist trce backend (dncNetworkName enc) genCfg (useShelleyInit enc)
-      liftIO $ epochStartup (enpExtended enp) trce backend
-
-    case genCfg of
-      GenesisCardano {} -> do
-        liftIO $ runSyncNodeClient metricsSetters syncEnv iomgr trce (enpSocketPath enp)
+      lift $ orDie renderSyncNodeError $ insertValidateGenesisDist trce backend (dncNetworkName syncNodeConfig) genCfg (useShelleyInit syncNodeConfig)
+      liftIO $ epochStartup (enpExtended syncNodeParams) trce backend
+    -- TODO: Vince: a lot of the downstream functions require LedgerEnv but is this right?
+    whenJust (envLedger syncEnv) $ \ledgerEnv ->
```

Comment:
I thought that was the case. But if there is no ledger directory or `--disable-leger` then `LedgerEnv` has to be `Nothing`. We either pull out the values that are used by both state (Has Ledger, Has no Ledger) which live inside `LedgerEnv` like `leTrace`. Because the only way you can get access to those is to take them out of `Maybe` but seems like they are used in places that don't require a `LedgerEnv`?